### PR TITLE
MAINT moved g-i-l* modules to pytest

### PR DESCRIPTION
**Pull request by NelleV**
This continues the migration from pytest to nose.

### CODE REVIEW

**Comment by NelleV** [comment]
@QuLogic @Kojoley What do you do with `from nose.plugins.attrib import attr`

File: lib/matplotlib/tests/test_legend.py [hunk 1]
```diff
@@ -1,14 +1,13 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import six
 from six.moves import xrange
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_legend.py]: If you want to get rid of this too, replace the `list(xrange(#))` by `np.arange(#)`; I don't think there's anything here that explicitly requires a list.

**Comment by QuLogic** [comment]
> @QuLogic @Kojoley What do you do with from nose.plugins.attrib import attr?

There's a plugin that tries to emulate the same test selection mechanism, but for basic equivalence, use `@pytest.mark.network` instead of `@attr('network')`.

File: lib/matplotlib/tests/test_gridspec.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import matplotlib.gridspec as gridspec
-from nose.tools import assert_raises, assert_equal
+from numpy.testing import assert_raises, assert_equal
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_gridspec.py]: So I've been testing it out without nose installed and `numpy.testing.assert_raises` doesn't work with nose installed. Can you replace it with the `pytest.raises` context manager?

File: lib/matplotlib/tests/test_image.py [hunk 3]
```diff
@@ -20,17 +20,11 @@
 import matplotlib.pyplot as plt
 
 from matplotlib import mlab
-from nose.tools import assert_raises
-from numpy.testing import (
-    assert_array_equal, assert_array_almost_equal, assert_allclose)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_image.py]: `assert_array_equal` is used in a few of the tests.

File: lib/matplotlib/tests/test_gridspec.py (hunk 2 above)
Comment (NelleV) [inline lib/matplotlib/tests/test_gridspec.py]: yep.
FYI, if you look at skimage's pytest transition PR, you'll have a good idea of what will work and what won't.
scikit-image/scikit-image#2468

File: lib/matplotlib/tests/test_legend.py [hunk 4]
```diff
@@ -59,9 +57,9 @@ def test_various_labels():
     # tests all sorts of label types
     fig = plt.figure()
     ax = fig.add_subplot(121)
-    ax.plot(list(xrange(4)), 'o', label=1)
+    ax.plot(np.arage(4), 'o', label=1)
     ax.plot(np.linspace(4, 4.1), 'o', label='D\xe9velopp\xe9s')
-    ax.plot(list(xrange(4, 1, -1)), 'o', label='__nolegend__')
+    ax.plot(np.arage(4, 1, -1), 'o', label='__nolegend__')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_legend.py]: `np.arange` in this one and the one above.

**Comment by soupault** [comment]
>This continues the migration from pytest to nose.

From `nose` to `pytest` you mean? :)

**Comment by phobson** [comment]
thanks for all of the merges, @jkseppan !

**Pull request merged**